eclair-core/src/main/scala/fr/acinq/eclair/router/Validation.scala

```diff
@@ -565,7 +626,7 @@ object Validation {
   def handleAvailableBalanceChanged(d: Data, e: AvailableBalanceChanged)(implicit log: LoggingAdapter): Data = {
     val (publicChannels1, graphWithBalances1) = e.shortIds.real.toOption.flatMap(d.channels.get) match {
       case Some(pc) =>
-        val pc1 = pc.updateBalances(e.commitments)
+        val pc1 = pc.updateBalances(e.commitments).copy(capacity = e.commitments.latest.capacity)
```

Comment:
That does look better, and thanks for the tests.

With this change and handling `WatchExternalChannelSpent` in `FundingTxWatcher` I'm getting closer to having the simple gossip integration working.

Reply:
Unfortunately I  couldn't find a way to test that the remote node (carol) updates their routes because the ChannelAnnoucement has the same scid after the splice and is ignored. This should be fixed by #2941 and the test can be updated then.